js/views/settingsVw.js

```diff
@@ -758,6 +787,7 @@ module.exports = Backbone.View.extend({
     if(newAddress.name || newAddress.street || newAddress.city || newAddress.state || newAddress.postal_code) {
       if(!newAddress.name || !newAddress.street || !newAddress.city || !newAddress.state || !newAddress.postal_code){
         messageModal.show(window.polyglot.t('errorMessages.saveError'), window.polyglot.t('errorMessages.missingError'));
+        setTimeout(() => $(e.target).removeClass('loading'), 0);
```

Comment:
I admit stopping propagation wasn't the cleanest solution. I looked into adding data to the event, so when it reaches the body there's a flag for whether to add the loading class or not. Can you think of a good way to do that?